services/jetbrains/jetbrains-downloads.service.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const Joi = require('joi')
+const JetbrainsBase = require('./jetbrains-base')
+const { metric } = require('../../lib/text-formatters')
+const {
+  downloadCount: downloadCountColor,
+} = require('../../lib/color-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schema = Joi.object({
+  'plugin-repository': Joi.object({
+    category: Joi.object({
+      'idea-plugin': Joi.array()
+        .min(1)
+        .items(
+          Joi.object({
+            '@_downloads': nonNegativeInteger,
+          })
+        )
+        .single()
```

Comment:
Is the single() needed here?

Reply:
When plugin has several versions (releases) `idea-plugin` is an array. If it has only one version - `idea-plugin` is an object. I use `single()` https://github.com/hapijs/joi/blob/v14.0.4/API.md#arraysingleenabled to allow single values and convert them to an array. We use `single()` in https://github.com/badges/shields/blob/d0fe97d136bbe981593c4821af44fa809eb14a86/services/ctan/ctan.service.js#L11 as well. In that case `license` is returned as a string or as an array.